var/spack/repos/builtin/packages/llvm-openmp-ompt/package.py

```diff
@@ -33,26 +33,69 @@ class LlvmOpenmpOmpt(CMakePackage):
 
     homepage = "https://github.com/OpenMPToolsInterface/LLVM-openmp"
 
-    # tr4-stable branch
+    # towards_tr4 branch
+    version('towards_tr4', branch='towards_tr4',
+            git='https://github.com/OpenMPToolsInterface/LLVM-openmp.git')
+
     version('3.9.2b2',
             git='https://github.com/OpenMPToolsInterface/LLVM-openmp.git',
             commit='5cdca5dd3c0c336d42a335ca7cff622e270c9d47')
+
     # align-to-tr-rebased branch
     version('3.9.2b',
             git='https://github.com/OpenMPToolsInterface/LLVM-openmp.git',
             commit='982a08bcf3df9fb5afc04ac3bada47f19cc4e3d3')
 
+    # variant for building llvm-openmp-ompt as a stand alone library
+    variant('standalone', default=False,
+            description="Build llvm openmpi ompt library as a \
+                         stand alone entity.")
+
+    # variant for building llvm-openmp-ompt using the toward_tr4 branch
+    variant('towardstr4', default=False,
```

Comment:
Why do u need a variant if it's already a version?

Reply:
@davydden Because I have to change the  omp version number for the build that corresponds to that branch in order to get the proper interface that we support and that this branch supports.
        # Build llvm-openmp-ompt using the toward_tr4 branch
        # This requires the version to be 5.0 (50)
        if '+towardstr4' in self.spec:
                cmake_args.extend(
                    ['-DLIBOMP_OMP_VERSION=50'])

Reply:
what I mean is that you can do
```
if '@towardstr4' in self.spec:
   ....// whatever

```